pageserver/src/metrics.rs

```diff
@@ -375,18 +394,73 @@ pub static WAL_REDO_RECORD_COUNTER: Lazy<IntCounter> = Lazy::new(|| {
     .unwrap()
 });
 
+pub struct StorageTimeMetricsTimer {
+    metrics: StorageTimeMetrics,
+    start: Instant,
+}
+
+impl StorageTimeMetricsTimer {
+    fn new(metrics: StorageTimeMetrics) -> Self {
+        Self {
+            metrics,
+            start: Instant::now(),
+        }
+    }
+
+    pub fn stop_and_record(self) {
+        let duration = self.start.elapsed().as_secs_f64();
+        self.metrics.timeline_sum.inc_by(duration);
+        self.metrics.timeline_count.inc();
+        self.metrics.global_histogram.observe(duration);
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct StorageTimeMetrics {
+    timeline_sum: Counter,
+    timeline_count: IntCounter,
+    global_histogram: Arc<Histogram>,
```

Comment:
```suggestion
    global_histogram: Histogram,
```

Similar to other counters from `prometheus` this is internally Arc'd.

Reply:
Removed.